direct-auth-samples/src/main/java/com/okta/spring/example/controllers/LoginController.java

```diff
@@ -0,0 +1,300 @@
+            return new ModelAndView("verify");
+        }
+
+        return new ModelAndView("login"); //TODO revisit this
+    }
+
+    @PostMapping("/verify")
+    public ModelAndView handleEmailVerify(@RequestParam("code") String code,
+                                          HttpSession httpSession) {
+        logger.info(":: Verify Code :: {}", code);
+
+        IDXClientContext idxClientContext = (IDXClientContext) httpSession.getAttribute("idxClientContext");
+
+        VerifyAuthenticatorOptions verifyAuthenticatorOptions = new VerifyAuthenticatorOptions();
+        verifyAuthenticatorOptions.setCode(code);
+
+        AuthenticationResponse authenticationResponse =
+                AuthenticationWrapper.verifyAuthenticator(client, idxClientContext, verifyAuthenticatorOptions);
+
+        if (authenticationResponse.getAuthenticationStatus() == AuthenticationStatus.AWAITING_PASSWORD_RESET) {
+            return new ModelAndView("change-password");
+        }
+
+        ModelAndView mav = new ModelAndView("login");
+        mav.addObject("messages", authenticationResponse.getAuthenticationStatus().toString());
+        return mav;
+    }
+
+    @PostMapping("/change-password")
+    public ModelAndView handleChangePassword(@RequestParam("new-password") String newPassword,
+                                             @RequestParam("confirm-new-password") String confirmNewPassword,
+                                             HttpSession httpSession) {
+        logger.info(":: Change Password ::");
+
+        if (!newPassword.equals(confirmNewPassword)) {
+            ModelAndView mav = new ModelAndView("change-password");
+            mav.addObject("errors", "Passwords do not match");
+            return mav;
+        }
+
+        ModelAndView mav = new ModelAndView("login");
+
+        IDXClientContext idxClientContext = (IDXClientContext) httpSession.getAttribute("idxClientContext");
+
+        ChangePasswordOptions changePasswordOptions = new ChangePasswordOptions();
+        changePasswordOptions.setNewPassword(newPassword);
+
+        AuthenticationResponse authenticationResponse =
+                AuthenticationWrapper.changePassword(client, idxClientContext, changePasswordOptions);
+
+        mav.addObject("info", authenticationResponse.getAuthenticationStatus().toString());
+        return mav;
+    }
+
+    @PostMapping("/register")
+    public ModelAndView handleRegister(@RequestParam("lastname") String lastname,
+                                       @RequestParam("firstname") String firstname,
+                                       @RequestParam("email") String email,
+                                       HttpSession session) {
+        logger.info(":: Register ::");
+
+        ModelAndView mav = new ModelAndView("enroll-authenticators");
+
+        NewUserRegistrationResponse newUserRegistrationResponse = AuthenticationWrapper.fetchSignUpFormValues(client);
```

Comment:
We should talk about direction with the team here. We aren't _really_ using this. Should we take a more dynamic approach, and fill in the UI from this? Or should we simplify the controller method, and allow some assumptions to be made, and do most of the logic in the wrapper. It would allow this integration point to be simplified a lot.

Reply:
Assumption here is we'll show the basic registration form UI with just three profile fields. This example is tied to that basic use case and the idea is to show the user how to build the UI with this assumption.